argilla/src/argilla/_helpers/_datasets_server.py

```diff
@@ -0,0 +1,175 @@
+    except KeyError as e:
+        raise DatasetsServerException() from e
+
+    available_configs = list(dataset_info.keys())
+
+    if len(available_configs) > 1 and config is None:
+        raise ValueError("Multiple configurations found. Please specify a config.")
+    elif len(available_configs) == 1:
+        config = available_configs[0]
+    elif config not in available_configs:
+        raise ValueError(f"Configuration '{config}' not found.")
+
+    try:
+        features = dataset_info[config]["features"]
+    except KeyError as e:
+        raise DatasetsServerException() from e
+
+    return features
+
+
+def _map_feature_type(feature):
+    """Map the feature type to the corresponding FeatureType enum."""
+
+    if isinstance(feature, list) and len(feature) > 0 and isinstance(feature[0], dict):
+        sub_features = feature[0]
+        if _is_chat_feature(sub_features):
+            return FeatureType.CHAT
+
+    hf_type = feature.get("_type")
+    dtype = feature.get("dtype")
+
+    if hf_type == "Value":
+        if dtype == "string":
+            return FeatureType.TEXT
+        elif dtype in ["int32", "int64"]:
+            return FeatureType.INT
+        elif dtype in ["float32", "float64"]:
+            return FeatureType.FLOAT
+    elif hf_type == "Image":
+        return FeatureType.IMAGE
+    elif hf_type == "ClassLabel":
+        return FeatureType.LABEL
+    else:
+        warnings.warn(f"Unsupported feature type. hf_type: {hf_type}, dtype: {dtype}")
+
+
+def _is_chat_feature(sub_features):
+    """Check if the sub_features correspond to an rg.ChatField."""
+    return (
+        "content" in sub_features
+        and "role" in sub_features
+        and sub_features["content"]["_type"] == "Value"
+        and sub_features["role"]["_type"] == "Value"
+        and sub_features["content"].get("dtype") == "string"
+        and sub_features["role"].get("dtype") == "string"
+    )
+
+
+def _define_settings_from_features(features: Union[List[Dict], Dict[str, Any]], config=None) -> "Settings":
+    """Define the argilla settings from the features of a dataset.
+
+    Parameters:
+        features (Dict[str, Dict[str, Any]): The features of the dataset.
+        config (str, optional): The configuration of the dataset. Defaults to None.
+
+    Returns:
+        rg.Settings: The settings defined from the features.
+    """
+
+    from argilla import ImageField, LabelQuestion, TextQuestion, Settings, TextField
+
+    fields = []
+    questions = []
+    metadata = []
+    mapping = {}
+
+    for name, feature in features.items():
+        feature_type = _map_feature_type(feature)
+
+        if feature_type == FeatureType.CHAT:
+            # TODO: Implement chat support to create `rg.ChatField`
+            # fields.append(rg.ChatField(name=f"{name}_field"))
+            # mapping[name] = f"{name}_field"
+            pass
+
+        elif feature_type == FeatureType.TEXT:
+            fields.append(TextField(name=f"{name}_field"))
```

Comment:
We discussed using values as both field and question so that it's as flexible as possible. Based on that, I need to differentiate the names for attributes and then map them.